bazel/envoy_library.bzl

```diff
@@ -66,6 +66,38 @@ EXTENSION_SECURITY_POSTURES = [
     "data_plane_agnostic",
 ]
 
+# Extension categories as defined by factories
+EXTENSION_CATEGORIES = [
+    "envoy.access_loggers",
+    "envoy.bootstrap",
+    "envoy.clusters",
+    "envoy.compression.compressor",
+    "envoy.compression.decompressor",
+    "envoy.filters.http",
+    "envoy.filters.listener",
+    "envoy.filters.network",
+    "envoy.filters.udp_listener",
+    "envoy.formatter",
+    "envoy.grpc_credentials",
+    "envoy.guarddog_actions",
+    "envoy.health_checkers",
+    "envoy.internal_redirect_predicates",
+    "envoy.io_socket",
```

Comment:
@htuch i have added `envoy.io_socket` category reflecting what was in the `source/extensions/extensions_build_config.bzl` file

afaict there is *not* a factory for it tho, not sure of the implications

Reply:
SGTM